derive_client/_bridge/w3.py

```diff
@@ -257,10 +272,16 @@ async def preflight_native_balance_check(
     total_cost = max_gas_cost + value
 
     if not balance >= total_cost:
+        chain_id = ChainID(await w3.eth.chain_id)
         ratio = balance / total_cost * 100
         raise InsufficientNativeBalance(
-            f"Insufficient funds: balance={balance}, required={total_cost} {ratio:.2f}% available "
-            f"(includes value={value} and assumed gas limit={ASSUMED_BRIDGE_GAS_LIMIT} at {max_fee_per_gas} wei/gas)"
+            f"Insufficient funds on {chain_id.name} ({chain_id}): "
+            f"balance={balance}, required={total_cost} {ratio:.2f}% available "
+            f"(includes value={value} and assumed gas limit={ASSUMED_BRIDGE_GAS_LIMIT} at {max_fee_per_gas} wei/gas)",
+            balance=balance,
+            chain_id=chain_id,
+            assumed_gas_limit=ASSUMED_BRIDGE_GAS_LIMIT,
+            fee_estimate=fee_estimate,
         )
```

Comment:
you want to intercept this exception and be able to match which chain this occurred on, and refund the EOA on this chain. Or, in case of absurd gas spikes or fee estimation, might decide to simply delay bridging for awhile and try again later.